File: src/main/java/dev/resmali/debugging/SmaliDebugAware.java
package dev.resmali.debugging;

import com.intellij.debugger.engine.JavaDebugAware;
import com.intellij.psi.PsiFile;
import org.jetbrains.annotations.NotNull;
import dev.resmali.SmaliLanguage;

public class SmaliDebugAware extends JavaDebugAware {
    @Override
    public boolean isBreakpointAware(@NotNull PsiFile psiFile) {
        return psiFile.getLanguage() == SmaliLanguage.INSTANCE;
    }
}

File: src/main/java/dev/resmali/debugging/SmaliDebugAware.kt
package dev.resmali.debugging

import com.intellij.debugger.engine.JavaDebugAware
import com.intellij.psi.PsiFile
import dev.resmali.SmaliLanguage

internal class SmaliDebugAware : JavaDebugAware() {
    override fun isBreakpointAware(psiFile: PsiFile): Boolean {
        return psiFile.language === SmaliLanguage.INSTANCE
    }
}
